tests/schema_evolution/code_gen/datatypes_remove_type/check.cpp

```diff
@@ -0,0 +1,33 @@
+#ifdef PODIO_SCHEMA_EVOLUTION_TEST_WRITE
+  #include "datamodel/ExampleHitRemovedCollection.h"
+#endif
+#include "datamodel/ExampleHitStaysCollection.h"
+
+#include "check_base.h"
+
+int main() {
+  WRITE_WITH(WriterT, (TEST_CASE FILE_SUFFIX), {
+    auto coll = ExampleHitRemovedCollection();
+    auto elem = coll.create();
+    elem.x(1.23);
+    elem.y(1.23);
+    elem.z(1.23);
+    elem.cellID(0xcaffee);
+    event.put(std::move(coll), "ExampleHitRemoved");
+    auto coll2 = ExampleHitStaysCollection();
+    auto elem2 = coll2.create();
+    elem2.t(99);
+    event.put(std::move(coll2), "evolution_collection");
+  });
+#ifdef PODIO_SCHEMA_EVOLUTION_TEST_READ
+  auto reader = ReaderT();
+  std::cout << "Reading file " << TEST_CASE FILE_SUFFIX << std::endl;
+  reader.openFile((TEST_CASE FILE_SUFFIX));
+  // TODO: Make this work when reading all collections
```

Comment:
Where / how does this currently fail? I thought we had the necessary bits and pieces in places also already for the `RNTupleReader` to simply not instantiate / populate the fields we don't want to read (?)

Reply:
Doesn't work with `ROOTReader` either at the moment, because we either read the full data or none at all. Let's fix this in another PR and move forward with this to unblock EDM4hep#446.

Reply:
I thought it was you who found out in https://github.com/key4hep/EDM4hep/pull/446 when only a subset of the collections is read. It certainly doesn't work right now.